### PR TITLE
feat(apt): add -f --fix-broken option to apt module

### PR DESCRIPTION
**Pull request by triat**
##### SUMMARY
Add the option to use --fix-broken from the apt packaging system with the state: fixed

##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
module/packaging/apt

##### ANSIBLE VERSION
```
ansible 2.3.1.0
  config file =
  configured module search path = Default w/o overrides
  python version = 2.7.13 (default, Dec 18 2016, 07:03:39) [GCC 4.2.1 Compatible Apple LLVM 8.0.0 (clang-800.0.42.1)]
```

##### ADDITIONAL INFORMATION

### CODE REVIEW

**Comment by triat** [comment]
Hey, do I need to do something more to the PR? Let me know if I can help you to integrate it easily. Thanks

**Comment by maxamillion** [comment]
@triat Thank you for the pull request! Apologies for the lag time on this, I'm currently going through the queue and trying to clean up stale PRs. I think the change looks fine but I would like to have some integration test(s) included. Thanks again.

**Comment by triat** [comment]
Hey @maxamillion! Don't worry, when I saw the 1600~ PR waiting, I was not expecting a quick answer. Thanks for taking the time to review this one, I'll make the test(s) ASAP.

**Comment by maxamillion** [comment]
@triat greatly appreciated, thanks again! :)

**Comment by mattclay** [comment]
CI failure in integration tests: https://app.shippable.com/github/ansible/ansible/runs/85811/39/tests

**Comment by triat** [comment]
Hey @mattclay,I just rebased on devel, pushed to upgrade the PR and saw it was not passing anymore. I have to check but I hadn't the time right now to figure out why I can not run the test on my mac (also in a linux VM as I saw that it is recommended to run the tests on a linux OS)

**Comment by mattclay** [comment]
@triat If you have Docker installed on your Mac you can run the test from your branch with:

`bin/ansible-test integration apt --docker ubuntu1604 -v`

File: lib/ansible/modules/packaging/os/apt.py [hunk 1]
```diff
@@ -1100,6 +1107,7 @@ def main():
                 force=force_yes,
                 dpkg_options=dpkg_options,
                 build_dep=state_builddep,
+                fixed=state_fixed,
```
Comment (mattclay) [inline lib/ansible/modules/packaging/os/apt.py:1111]: The tests are failing because `state_fixed` is undefined when `p['state'] != 'fixed'`.

Comment (triat) [reply lib/ansible/modules/packaging/os/apt.py:1111]: thanks for the check, that's actually pretty obvious looking at the code I've made!

**Comment by triat** [comment]
I've been trying this command but I don't know what I've done to get it wrong. I ran it again and it was magically properly failing. Anyway, thanks again for the help, I've fixed the initialization and I'll have to write a test now.

Regarding this part, is there any recommendation if I need to install something with `dpkg` which will miss some dependencies that I'll then install through `apt -f` ?

**Comment by maxamillion** [comment]
rebuild_merge

**Pull request merged**